lib/strutil.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * String helper routines
+ *
+ * Copyright 2018, 2023 Phoenix Systems
+ * Author: Jan Sikorski, Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "../hal/hal.h"
+#include "../vm/kmalloc.h"
+
+
+static char *lib_strrchr(const char *s, int c)
+{
+	const char *p = NULL;
+
+	do {
+		if (*s == c) {
+			p = s;
+		}
+	} while (*(s++) != '\0');
```

Comment:
is incremetn in while condition compliant with misra 13.3?

Reply:
I'd rewrite it using for loop for clarity

Reply:
Done